File: 프로그래머스/lv1/12977. 소수 만들기/소수 만들기.java
class Solution {
    public int solution(int[] nums) {
        int answer = 0;

        for(int i=0; i<nums.length; i++) {
            for(int j=i+1; j<nums.length; j++) {
                for(int k=j+1; k<nums.length; k++) {
                    int num = nums[i] + nums[j] + nums[k];
                    if(isPrime(num)) answer++;
                }
            }
        }

        return answer;
    }
    
    public boolean isPrime(int num) {
        for (int j = 2; j * j <= num; j++) {
            if (num % j == 0) return false;
        }
        
        return true;
    }
}

File: 프로그래머스/1/12977. 소수 만들기/소수 만들기.js
function solution(nums) {
    let answer = 0;

    for(let i=0; i<nums.length; i++) {
        for(let j=i+1; j<nums.length; j++) {
            for(let k=j+1; k<nums.length; k++) {
                if(isPrime(nums[i] + nums[j] + nums[k])) answer++;
            }
        }
    }

    return answer;
}
    
function isPrime(num) {
    for (let j = 2; j * j <= num; j++) {
        if (num % j == 0) return false;
    }
        
    return true;
}
